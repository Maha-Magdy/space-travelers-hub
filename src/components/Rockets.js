/* eslint-disable react/prop-types */
import axios from 'axios';
import { useSelector, useDispatch, connect } from 'react-redux';
import { useEffect } from 'react';
import { rocketsList } from '../redux/rockets/rockets';
import Rocket from './Rocket';

function Rockets() {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocketsReducer);

  useEffect(() => {
    if (rockets.length === 0) {
      axios.get('https://api.spacexdata.com/v3/rockets').then((response) => {
        const { data } = response;
        const result = data.map((rocket) => ({
          id: rocket.rocket_id,
          name: rocket.rocket_name,
          description: rocket.description,
          flickr_images: rocket.flickr_images,
        }));
        dispatch(rocketsList(result));
      });
    }
  }, [dispatch, rockets.length]);

  if (rockets !== []) {
    return (
      <ul className="container-fluid rockets">
        {rockets.map((rocket) => (
          <li key={rocket.id} className="mb-3">
            <Rocket rocket={rocket} />
          </li>
        ))}
      </ul>
    );
  }

  return <h1>Here is Rockets page</h1>;
}

const mapStateToProps = (state) => ({
  rockets: state.rocketsReducer,
});

export default connect(mapStateToProps)(Rockets);
