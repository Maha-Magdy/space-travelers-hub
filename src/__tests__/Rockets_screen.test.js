import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';

it('Rockets screen renders as expected', () => {
  const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
