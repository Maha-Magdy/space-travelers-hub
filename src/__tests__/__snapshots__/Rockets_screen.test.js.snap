// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets screen renders as expected 1`] = `
<ul
  className="container-fluid rockets"
/>
`;
